### ISSUE TITLE
Level editor?

### ISSUE THREAD

**Issue by bates64** [liam4/slw#26]
We've been hinting at this for a while - we _need_ a proper level editor for SLW. Editing a single tilemap layer in `src/levels.toml` is going to get _really_ ugly and it'd be nice to have anyway.

And as we all know, I :heart: polish, so if we can make the editor Super Mario Maker-like, it'd be awesome :smile:

**Comment by towerofnix** [comment]
Having multiple layers is good.

In terms of editing multiple layers - perhaps have a way to review layers by having everything on the selected layer shake a little, and have all other layers ~~hidden~~ dimmed?

**Comment by bates64** [comment]
Actually, I'm curious - @liam4 have you ever played SMM?

How about make other layers fade slightly rather than totally hide them? :+1: to the shaking though.

**Comment by towerofnix** [comment]
Nope! That suggestion took inspiration from [OHRRPGCE](http://rpg.hamsterrepublic.com/ohrrpgce/Main_Page) though.

Oh oops. I meant to say dimmed.

**Comment by bates64** [comment]
Play it! Or at least _watch_ some of the level editor gameplay. It uses shakes and sounds to its advantage very well to do stuff.

**Comment by an-OK-squirrel** [comment]
can we please be able to have foreground blocks on the background I love that mechanic in nsmbw

**Comment by bates64** [comment]
@an-OK-squirrel Foreground blocks? What, like tiles that are normally solid in the background?

**Comment by bates64** [comment]
Unless you mean the "hidden area" mechanic.

**Comment by PullJosh** [comment]
I assume enemies will be a part of slw. Is there a plan for handling this?

**Comment by towerofnix** [comment]
[Mario Maker is cool.](https://www.youtube.com/watch?v=7pIDgaBJND8)

**Comment by towerofnix** [comment]
> I assume enemies will be a part of slw. Is there a plan for handling this?

No _concrete_ plan, but yeah, that's obviously an important thing to have that will come.

BTW, there's already goombas (`G` tile). But I don't know if they still work, I haven't tested them in a while.

**Comment by bates64** [comment]
Goombas are basically mushrooms with "when jumped on" events and a different `onTouch` handler :wink: 

We need sprites for enemies though.
